Userland/Applications/PartitionEditor/main.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2022, Samuel Bowman <sam@sambowman.tech>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Applications/PartitionEditor/PartitionEditorWindowGML.h>
+#include <Applications/PartitionEditor/PartitionModel.h>
+#include <LibCore/DirIterator.h>
+#include <LibCore/System.h>
+#include <LibGUI/Application.h>
+#include <LibGUI/ComboBox.h>
+#include <LibGUI/ItemListModel.h>
+#include <LibGUI/Menu.h>
+#include <LibGUI/MessageBox.h>
+#include <LibGUI/TableView.h>
+
+using namespace PartitionEditor;
+
+Vector<String> get_device_paths();
+
+Vector<String> get_device_paths()
+{
+    auto device_paths = Vector<String>();
+    Core::DirIterator iterator("/dev", Core::DirIterator::SkipParentAndBaseDir);
+    while (iterator.has_next()) {
+        auto path = iterator.next_full_path();
+        if (Core::File::is_block_device(path))
+            device_paths.append(path);
+    }
+    return device_paths;
+}
+
+ErrorOr<int> serenity_main(Main::Arguments arguments)
+{
+    TRY(Core::System::pledge("stdio recvfd sendfd rpath unix"));
+
+    auto app = TRY(GUI::Application::try_create(arguments));
+
+    TRY(Core::System::pledge("stdio recvfd sendfd rpath"));
+    TRY(Core::System::unveil("/dev", "r"));
+    TRY(Core::System::unveil("/res", "r"));
+    TRY(Core::System::unveil(nullptr, nullptr));
+
+    // FIXME: PartitionEditor needs its own icon.
+    auto app_icon = TRY(GUI::Icon::try_create_default_icon("app-space-analyzer"));
+
+    auto window = TRY(GUI::Window::try_create());
+    window->set_title("Partition Editor");
+    window->resize(640, 400);
+    window->set_icon(app_icon.bitmap_for_size(16));
+
+    // FIXME: Abort and show a dialog if not running as root.
```

Comment:
That doesn't seem like great UX. Instead we should have functionality to prompt the user for a password to authenticate as root once needed (this may already exist; not sure).

...this is assuming that read-only use doesn't need root :)

Reply:
I 100% agree. This is not great UX and we should prompt the user to authenticate as root. Unfortunatly, AFAIK no such functionality exists for GUI apps yet (CLI has pls). Without this message box, attempting to run as a non-root user silently fails with a permission error, which is really bad UX.